src/stack_aggregator.py

```diff
@@ -19,6 +19,30 @@
 logger = logging.getLogger(__file__)
 
 
+def get_recommended_version(ecosystem, name, version):
+    """Fetch the recommended version in case of CVEs."""
+    query = "g.V().has('ecosystem', '{eco}').has('name', '{pkg}')" \
+            ".out('has_version').not(out('has_cve')).values('version');"\
+        .format(eco=ecosystem, pkg=name)
+    payload = {'gremlin': query}
+    result = execute_gremlin_dsl(url=GREMLIN_SERVER_URL_REST, payload=payload)
+    if result:
+        versions = result['result']['data']
+        if len(versions) == 0:
+            return None
+    else:
+        return None
+    rec_version = version
+    for ver in versions:
+        rec_version = select_latest_version(
```

Comment:
I don't understand this logic here. You are overwriting rec_version every time. What if the list of versions is
['1.4', '1.7', '1.3', '1.6', '1.5'] and user's version is '1.5' with CVE. What will happen here? Are we sending None or 1.7?

Reply:
the rec_version will have the latest value everytime.
Example : user's version is 1.6 and the list we get is as you mentioned above, here is how it will calculate in each iteration
iterations
1. 1.6 > 1.4 -> 1.6 [ in the rec_version]
2. 1.7
3. 1.7
4. 1.7
5. 1.7

Reply:
Ok thanks